File: src/STOTOP.Module.Finance/AccountSetPermissions.cs
namespace STOTOP.Module.Finance;

/// <summary>
/// 账套级权限编码常量
/// </summary>
public static class AccountSetPermissions
{
    // 凭证权限
    public const string VoucherView = "accountset:voucher:view";
    public const string VoucherCreate = "accountset:voucher:create";
    public const string VoucherEdit = "accountset:voucher:edit";
    public const string VoucherDelete = "accountset:voucher:delete";
    public const string VoucherSubmit = "accountset:voucher:submit";
    public const string VoucherAudit = "accountset:voucher:audit";
    public const string VoucherUnaudit = "accountset:voucher:unaudit";
    public const string VoucherPrint = "accountset:voucher:print";

    // 期间权限
    public const string PeriodView = "accountset:period:view";
    public const string PeriodClose = "accountset:period:close";
    public const string PeriodReopen = "accountset:period:reopen";

    // 报表权限
    public const string ReportView = "accountset:report:view";
    public const string ReportExport = "accountset:report:export";

    // 科目权限
    public const string SubjectView = "accountset:subject:view";
    public const string SubjectEdit = "accountset:subject:edit";

    // 辅助核算权限
    public const string AuxiliaryView = "accountset:auxiliary:view";
    public const string AuxiliaryEdit = "accountset:auxiliary:edit";

    // 账套设置权限
    public const string SettingsView = "accountset:settings:view";
    public const string SettingsEdit = "accountset:settings:edit";

    // 出纳日记账权限
    public const string CashJournalView = "accountset:cashjournal:view";
    public const string CashJournalEdit = "accountset:cashjournal:edit";

    // 账套授权管理
    public const string AuthorizationManage = "accountset:authorization:manage";
}

File: web/src/constants/accountSetPermissions.ts
/**
 * 账套级权限编码常量
 */
export const AccountSetPermissions = {
  // 凭证
  VoucherView: 'accountset:voucher:view',
  VoucherCreate: 'accountset:voucher:create',
  VoucherEdit: 'accountset:voucher:edit',
  VoucherDelete: 'accountset:voucher:delete',
  VoucherSubmit: 'accountset:voucher:submit',
  VoucherAudit: 'accountset:voucher:audit',
  VoucherUnaudit: 'accountset:voucher:unaudit',
  VoucherPrint: 'accountset:voucher:print',
  // 期间
  PeriodView: 'accountset:period:view',
  PeriodClose: 'accountset:period:close',
  PeriodReopen: 'accountset:period:reopen',
  // 报表
  ReportView: 'accountset:report:view',
  ReportExport: 'accountset:report:export',
  // 科目
  SubjectView: 'accountset:subject:view',
  SubjectEdit: 'accountset:subject:edit',
  // 辅助核算
  AuxiliaryView: 'accountset:auxiliary:view',
  AuxiliaryEdit: 'accountset:auxiliary:edit',
  // 设置
  SettingsView: 'accountset:settings:view',
  SettingsEdit: 'accountset:settings:edit',
  // 出纳日记账
  CashJournalView: 'accountset:cashjournal:view',
  CashJournalEdit: 'accountset:cashjournal:edit',
} as const
